app/models/katello/glue/candlepin/owner.rb

```diff
@@ -16,8 +16,19 @@ def self.included(base)
     end
 
     module InstanceMethods
+      def find_owner
+        Resources::Candlepin::Owner.find(self.label)
+      end
+
+      def candlepin_owner_exists?
+        find_owner
+        true
+      rescue RestClient::ResourceNotFound
```

Comment:
this method is a tad dangerous as it cannot detect between candlepin being down and the owner being non-existant.  Dynflow would 'ping' candlepin before the candlepin action, i wonder if this should too (instead of adding it everywhere this is called).  It might result in duplicates but these should be quick calls i think?

Reply:
Watcha think about pinging candlepin in the `OrganizationCreator` and instead of checking if the owner exists (assuming we get this far ie candlepin is up) we just try to create the owner and rescue the BadRequest if it already exists?

Reply:
does it return badrequest? or 409 conflict?    If we can detect a conflict over  a bad request i'd maybe be okay with it.  But if we get back a  bad request, it might be bad to assume it exists?

Reply:
You can also ping in the `rescue` if the 404 doesn't contain a guarantee from Candlepin. I don't know Candlepin well enough, but perhaps it always sends a header that you can check that it does indeed come from the application. If not, it could be an interesting conversation to have with upstream.

Reply:
Thanks @ekohl for the header check inspiration. I went that route.

Take a look at the latest commit and see how the error handling looks @jlsherrill  - I think it's safer now

Reply:
I like this approach!